### ISSUE TITLE
Add support for oblivious DNS

### ISSUE THREAD

**Issue by JayNewstrom** [qdm12/dns#51]
https://blog.cloudflare.com/oblivious-dns/

**Comment by qdm12** [comment]
Yes I've been eyeing this to jump from DNS over TLS to this, instead of going DNS over HTTPS which adds little benefit apart from using port 443. It might take some time to implement as a local DNS relay (as Unbound currently does) but it is definitely in my brain roadmap 😉

**Comment by JayNewstrom** [comment]
Sounds good. Let me know if you need help testing :)

**Comment by qdm12** [comment]
Still a bit behind oblivious DNS, but I made good progress on a Go implementation for a DNS over HTTPs and DNS over TLS proxying server https://github.com/qdm12/dns/pull/58  it should be finalized in the coming week.

Oblivious dns would be next on the list! 😉

**Comment by qdm12** [comment]
Note to future self: https://github.com/cloudflare/odoh-client-go should be a good starting point
